diff --git a/Makefile b/Makefile
index 9522b89a7..42b9caf91 100644
--- a/Makefile
+++ b/Makefile
@@ -36,7 +36,7 @@ $(CMD_BINS): build_cmds
 
 build_cmds: | $(OBJDIR)
 	echo $(OBJECTS)
-	GOBIN=$(OBJDIR) GO111MODULE=on go install -mod=vendor $(GO_BUILD_FLAGS) ./...
+	GOBIN=$(GOBIN) GO111MODULE=on go install -mod=vendor $(GO_BUILD_FLAGS) ./...
 
 # Building a .deb requires `fpm` from https://github.com/jordansissel/fpm
 # which you can install with `gem install fpm`.
diff --git a/cmd/ceremony/cert.go b/cmd/ceremony/cert.go
index 7f3893593..f32abfeba 100644
--- a/cmd/ceremony/cert.go
+++ b/cmd/ceremony/cert.go
@@ -69,6 +69,12 @@ var AllowedSigAlgs = map[string]x509.SignatureAlgorithm{
 	"ECDSAWithSHA256": x509.ECDSAWithSHA256,
 	"ECDSAWithSHA384": x509.ECDSAWithSHA384,
 	"ECDSAWithSHA512": x509.ECDSAWithSHA512,
+	"SHA3_256WithRSA":   x509.SHA3_256WithRSA,
+	"SHA3_384WithRSA":   x509.SHA3_384WithRSA,
+	"SHA3_512WithRSA":   x509.SHA3_512WithRSA,
+	"ECDSAWithSHA3_256": x509.ECDSAWithSHA3_256,
+	"ECDSAWithSHA3_384": x509.ECDSAWithSHA3_384,
+	"ECDSAWithSHA3_512": x509.ECDSAWithSHA3_512,
 }
 
 type certType int
diff --git a/pkcs11helpers/helpers.go b/pkcs11helpers/helpers.go
index 173123e17..7d74466b5 100644
--- a/pkcs11helpers/helpers.go
+++ b/pkcs11helpers/helpers.go
@@ -239,6 +239,10 @@ var hashIdentifiers = map[crypto.Hash][]byte{
 	crypto.SHA256: {0x30, 0x31, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x01, 0x05, 0x00, 0x04, 0x20},
 	crypto.SHA384: {0x30, 0x41, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x02, 0x05, 0x00, 0x04, 0x30},
 	crypto.SHA512: {0x30, 0x51, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x03, 0x05, 0x00, 0x04, 0x40},
+	// SHA3 Indetifiers
+	crypto.SHA3_256: {0x30, 0x31, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x08, 0x05, 0x00, 0x04, 0x20},
+	crypto.SHA3_384: {0x30, 0x41, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x09, 0x05, 0x00, 0x04, 0x30},
+	crypto.SHA3_512: {0x30, 0x51, 0x30, 0x0d, 0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x0A, 0x05, 0x00, 0x04, 0x40},
 }
 
 func (s *Session) Sign(object pkcs11.ObjectHandle, keyType keyType, digest []byte, hash crypto.Hash) ([]byte, error) {
